src/base/static/state/ducks/map.js

```diff
@@ -0,0 +1,107 @@
+// Selectors
+export const mapBasemapSelector = state => {
+  return state.map.visibleBasemapId;
+};
+export const mapLayersStatusSelector = state => {
+  return state.map.layersStatus;
+};
+export const mapSizeValiditySelector = state => {
+  return state.map.isMapSizeValid;
+};
+export const mapPositionSelector = state => {
+  return state.map.mapPosition;
+};
+
+// Actions
+const SET_LAYER_STATUS = "map/SET_LAYER_STATUS";
+const SET_BASEMAP_STATUS = "map/SET_BASEMAP_STATUS";
+const SET_MAP_POSITION = "map/SET_MAP_POSITION";
+const SET_MAP_SIZE_VALIDITY = "map/SET_MAP_SIZE_VALIDITY";
+
+// Action creators
+export const setLayerStatus = (layerId, layerStatus) => {
+  return {
+    type: SET_LAYER_STATUS,
+    payload: {
+      layerId: layerId,
+      layerStatus: layerStatus,
+    },
+  };
+};
+export const setBasemap = (layerId, layerStatus) => {
+  return {
+    type: SET_BASEMAP_STATUS,
+    payload: {
+      layerId: layerId,
+      layerStatus: layerStatus,
+    },
+  };
+};
+export const setMapPosition = mapPosition => {
+  return {
+    type: SET_MAP_POSITION,
+    payload: mapPosition,
+  };
+};
+export const setMapSizeValidity = isValidSize => {
+  return {
+    type: SET_MAP_SIZE_VALIDITY,
+    payload: isValidSize,
+  };
+};
+
+// Reducers
+const INITIAL_STATE = {
+  visibleBasemapId: null,
+  layersStatus: {},
```

Comment:
To me it makes sense to merge the two. So a "first-class" layer would consist of information in the `layers` section of the config, plus to "status" properties: `isVisible` and `status`.

Reply:
Using the id as the hash key is convenient for quick lookups. Maybe we could replicate the key in the payload too? So we'd have something like this:
```
"layer-a": {
  id: "layer-a",
  status: "loading",
  isBasemap: false,
  type: "vector"
}
```

Reply:
As discussed, merging these layers seems reasonable. It's also pretty trivial, so we can hold off on it for now.

>Using the id as the hash key is convenient for quick lookups
Do you think we would have enough layers where this would be a concern? I'm fine with either representation, it just seems that an array of objects is simpler.

>Maybe we could replicate the key in the payload too?
I think that would make sense.